source/views/tasks_view.py

```diff
@@ -38,15 +39,18 @@ def post(self):
         data = request.json
         task_name = data.get('task_name', None)
         if not task_name:
-            return 400
+            raise NotFound('Task name is None')
         task_description = data.get('task_description', None)
-        tsk = Task(task_name=task_name, task_description=task_description)
+        due_date = data.get('due_date', None)
+        if due_date is not None:
+            due_date = datetime.fromtimestamp(int(due_date)).strftime('%Y-%m-%d %H:%M:%S')
+        tsk = Task(task_name=task_name, task_description=task_description, due_date=due_date)
         try:
             db.session.add(tsk)
             db.session.commit()
         except Exception as e:
             db.session.rollback()
-            return 409
+            raise Conflict('Conflict adding task')
```

Comment:
Everything that goes in an `except` block when the code in the try is out of your scope (like in this case is all sqlalchemy stuff) you can't be sure of the reason, and certaintly is not `Conflict` since you don't know that the error happens because of a database conflict of values or primary keys. When this happens, you raise `InternalServerError`